### ISSUE TITLE
Problems with the code for texture synthesis 

### ISSUE THREAD

**Issue by cchen156** [anishathalye/neural-style#18]
Hi,

First, thanks a lot for your code! 

I tried this code for texture synthesis by setting the content weight to zero. The result is much worse than that by the torch code: https://github.com/jcjohnson/neural-style

I finally figure out the issues. In the vgg network, the max pooling should be replaced by the average pooling. This helps a lot! Also, the TV loss should be L1 norm of the gradients instead of L2 norm. It improves the quality. 

Thanks!

**Comment by anishathalye** [comment]
Hmm, it looks like jcjohnson's implementation uses max pooling by default.

Also, why should TV loss be an L1 norm instead of an L2 norm?

**Comment by cchen156** [comment]
L1 norm will induce sharper edges of the images. See the definitions: https://en.wikipedia.org/wiki/Total_variation_denoising

**Comment by anishathalye** [comment]
Are you interested in submitting a pull request?

**Comment by cchen156** [comment]
Just a few lines changed:

        '''
        tv_loss = tv_weight * 2 * (
                (tf.nn.l2_loss(image[:,1:,:,:] - image[:,:shape[1]-1,:,:]) /
                    tv_y_size) +
                (tf.nn.l2_loss(image[:,:,1:,:] - image[:,:,:shape[2]-1,:]) /
                    tv_x_size))
        '''
        
        tv_y  = tf.reduce_sum(tf.abs(image[:,1:,:,:] - image[:,:shape[1]-1,:,:])) / tv_y_size
        tv_x  = tf.reduce_sum(tf.abs(image[:,:,1:,:] - image[:,:,:shape[2]-1,:])) / tv_x_size
        tv_loss = tv_weight  * (tv_x + tv_y)

**Comment by anishathalye** [comment]
Hmm, I played around a little bit with using L1 loss instead of L2 loss, at least for style transfer, and the result doesn't look any better to me (in fact, it looks a little worse on the examples I tried).

I tried replacing max pooling with avg pooling as well, and the result looks significantly worse.
